### DISCUSSION TITLE
ts-node can't find basis configuration

### DISCUSSION THREAD

**Discussion by bkerin** [TypeStrong/ts-node#1895] [Q&A]
I wanted to set up my own dialect of ts to be used by default for free scripts (no tsconfig.json etc.), ts-node looks good, so:

` #!/usr/bin/env -S ts-node --esm --pretty --project="${HOME}/.my.tsconfig.json"`

where `~/.my.tsconfig.json` specifies my dialect.

One thing I've run into is that it doesn't seem to resolve some things especially where tsc is involved, for example with 

`"extends":  "@tsconfig/node18-strictest-esm/tsconfig.json",`
 
in ~/.my.tsconfig.json I get an error:

```
TSError: ⨯ Unable to compile TypeScript:
error TS6053: File '@tsconfig/node18-strictest-esm/tsconfig.json' not found.
```

replacing the link with the full path the that tsconfig.json works.

Is there some option that can be passed through to tsc to make this work (and improve module resolution generally)?

I tried  `--compilerOptions "{\"baseUrl\": \"${NODE_PATH}\"}" ` but still got the same error.  I notice giving a bogus compiler option name (e.g. baaaaseUrl) still gives the same error so it doesn't seem like this option is making to the compiler at all,
do I have it wrong somehow? 

I get the impression from this discussion that maybe environmental ts just isn't really done yet generally:

https://github.com/TypeStrong/ts-node/discussions/1450

**Comment by bkerin** [comment]
https://www.npmjs.com/package/tsconfig-paths also doesn't seem to help with this

**Comment by cspotcode** [comment]
Have you tested how `tsc` behaves when trying to resolve the exact same `"extends"` config from the exact same config file at the exact same location on your filesystem?

Something like this:

```
tsc --project $HOME/.my.tsconfig.json --showConfig
```

Whatever output you receive as output, be sure to *interpret* it and determine what it means.  Some errors may indicate success.

**Comment by bkerin** [comment]
I get the impression that node/ts/etc. aren't really supported without packages.json tsconfig.json and node_modules/ everywhere (reasons below).  Is this a fair statement?  Is there any real intention to improve the situation?

It's a bit sad since ts-node has been downloaded 87M times or so and at least some of those must be interested in shebang like old times but maybe this space is just left for python still.

Reasons:

* ts-node says "Via tsconfig.json (recommended)"
* things said in https://nodejs.org/api/modules.html#loading-from-the-global-folders
* npm install foo without packages.json/./node_modules/ undocumentedly and silently stuffs things in ~/node_modules if you're under ~, but current dir if under /tmp (surprise!)
* npm install -g doesn't seem to end up with modules that tsc at least actually know how to find, and tsc appears to offer ~0 support for free ts files (despite some minimal support in the language server)



Well I couldn't reproduce the behavior I thought I was seeing.  I've tried pretty hard now but can't get ts-node to behave reasonably without packages.json and node_modules/.

```
$ tsc --project $HOME/.my.tsconfig.json --showConfig

../../../.my.tsconfig.json:20:15 - error TS6053: File '@tsconfig/node18-strictest-esm/tsconfig.json' not found.

20   "extends":  "@tsconfig/node18-strictest-esm/tsconfig.json",
                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
